internal/venice-client-common/src/main/java/com/linkedin/venice/schema/writecompute/WriteComputeHandlerV1.java

```diff
@@ -112,6 +162,55 @@ Object updateArray(Schema arraySchema, List originalArray, Object writeComputeAr
     return originalArray;
   }
 
+  Object mergeArray(Schema arraySchema, Object currArrayUpdate, Object newArrayUpdate) {
+    if (getFieldOperationType(newArrayUpdate).equals(PUT_NEW_FIELD)) {
+      return newArrayUpdate; // Partial update on a list field
+    }
+    /**
+     * If old array update is {@link LIST_OPS}, merge it into the new collection merge OP's {@link SET_UNION} field.
+     * If old array update is NULL, a new empty array is created to take in the new collection merge.
+     * Otherwise, they are both {@link LIST_OPS}, just merge both {@link SET_UNION} field and {@link SET_DIFF} field.
+     */
+    List newSetUnion = (List) ((GenericRecord) newArrayUpdate).get(SET_UNION);
+    List newSetDiff = (List) ((GenericRecord) newArrayUpdate).get(SET_DIFF);
+    if (getFieldOperationType(currArrayUpdate).equals(LIST_OPS)) {
+      List currSetUnion = (List) ((GenericRecord) currArrayUpdate).get(SET_UNION);
+      List currSetDiff = (List) ((GenericRecord) currArrayUpdate).get(SET_DIFF);
+      // Creating a new updated list to avoid the case that old list is unmodifiable.
+      List updatedSetUnion = new GenericData.Array(arraySchema, newSetUnion);
+      List updatedSetDiff = new GenericData.Array(arraySchema, newSetDiff);
+      for (Object element: currSetUnion) {
+        if (!updatedSetUnion.contains(element)) {
+          updatedSetUnion.add(element);
+        }
+      }
+      for (Object elementToRemove: currSetDiff) {
+        if (!updatedSetDiff.contains(elementToRemove) && !updatedSetUnion.contains(elementToRemove)) {
+          updatedSetDiff.add(elementToRemove);
+        }
+      }
```

Comment:
hmm how are we handling the case setUnion:[A, B] followed by setDiff[B]? If I'm reading this code correctly the updated setUnion will be [A, B] and updated setDiff will be [] because of `!updatedSetUnion.contains(elementToRemove)` and that's incorrect right? The expected result should be setUnion[A] setDiff[]. So if we get another setUnion [B] we can consolidate it to setUnion[A, B] again.

Reply:
Discussed offline, this won't be an issue.